main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala

```diff
@@ -625,6 +625,21 @@ object SemanticTokensProvider {
   private def visitExps(exps0: List[Expr]): Iterator[SemanticToken] =
     exps0.flatMap(visitExp).iterator
 
+  private def visitCst(constant: Expr.Cst): Iterator[SemanticToken] = constant.cst match {
+    case _ : Constant.Str |
+         _ : Constant.Char => Iterator(SemanticToken(SemanticTokenType.String, Nil, constant.loc))
+    case _ : Constant.Int8 |
+         _ : Constant.Int16 |
+         _ : Constant.Int32 |
+         _ : Constant.Int64 |
+         _ : Constant.BigInt |
+         _ : Constant.Float32 |
+         _ : Constant.Float64 |
+         _ : Constant.BigDecimal => Iterator(SemanticToken(SemanticTokenType.Number, Nil, constant.loc))
+    case _: Constant.Regex => Iterator(SemanticToken(SemanticTokenType.Regexp, Nil, constant.loc))
+    case _ => Iterator(SemanticToken(SemanticTokenType.Type, Nil, constant.loc))
```

Comment:
In particular the wildcard case should avoided: Otherwise we won't get errors when the compiler is extended with a new constant.